src/main/kotlin/no/skatteetaten/aurora/gobo/integration/dbh/databaseServices.kt

```diff
@@ -112,39 +117,46 @@ class DatabaseSchemaServiceReactive(
         return webClient
             .post()
             .uri("/api/v1/schema/")
-            .body(BodyInserters.fromObject(input))
-            .header(HttpHeaders.AUTHORIZATION, "$HEADER_AURORA_TOKEN ${sharedSecretReader.secret}")
+            .body(BodyInserters.fromValue(input))
+            .authHeader()
             .retrieve()
             .bodyToMono<DbhResponse<*>>()
             .item()
     }
 
-    private fun Mono<DbhResponse<*>>.item() = this.items().map { it.first() }
+    private fun WebClient.RequestHeadersSpec<*>.authHeader() =
+        this.header(HttpHeaders.AUTHORIZATION, "$HEADER_AURORA_TOKEN ${sharedSecretReader.secret}")
+
+    private inline fun <reified T : Any> Mono<DbhResponse<*>>.item(): Mono<T> = this.items<T>().map { it.first() }
 
-    private fun Mono<DbhResponse<*>>.items() =
+    private inline fun <reified T : Any> Mono<DbhResponse<*>>.items(): Mono<List<T>> =
```

Comment:
Kan du bruke switchOnEmpty her?

Reply:
Jeg skjønte ikke denne metoden helt i starten, derfor foreslo jeg det som stod over. onSuccesss on onFailure metodene er veldig like standard metodene som er på en mono, jeg trodde man mente disse her.

Reply:
Jeg synes dette er et godt eksempel på hvor "konvolutten" vår har noen mangler. Siden vi bruker items listen som både error og success så er det vanskelig å marshalle hele til en payload i en smell med den innebygde decoderen i spring. 

Hvis konvolutten ikke hadde hatt success:Boolean, items:List<T> men i steden for items:List<A>, errors:List<E> så ville det vært mulig å sende med inn Response<DBhResponse, DbhError> som type for å marshalle opp objektet.

Reply:
Det er jeg helt enig i, at vi bruker items til ulike prosjekter gjør ting litt vanskeligere enn det trengere å være. Noe å fikse, men kanskje i en annen task?